CMakeLists.txt

```
@@ -0,0 +1,44 @@
+cmake_minimum_required(VERSION 2.8)
+project( Gazo-san )
+set (CMAKE_CXX_STANDARD 11)
+set(GTEST OFF CACHE BOOL "Test flag")
+find_package( OpenCV REQUIRED )
+include_directories( ${OpenCV_INCLUDE_DIRS} )
+
+if(GTEST)
+  set (CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR})
+  set (gtest ${CMAKE_SOURCE_DIR}/tests/googletest/libgtest.a)
+  set (gtest_main ${CMAKE_SOURCE_DIR}/tests/googletest/libgtest_main.a)
+  include_directories( src/ )
+  include_directories( include/ )
+  include_directories( tests/googletest/include/ )
+  file(DOWNLOAD https://raw.githubusercontent.com/jarro2783/cxxopts/master/include/cxxopts.hpp ${CMAKE_SOURCE_DIR}/include/cxxopts.hpp)
+
+  if( ${TEST_LEVEL} STREQUAL "unit" )
+    add_executable(gazosan_unit_test tests/unit_tests/imageDiffCalc_test.cpp)
+    target_link_libraries( gazosan_unit_test ${OpenCV_LIBS} ${gtest} ${gtest_main} -lpthread )
```

I want to commonalize.

```suggestion
    target_link_libraries( gazosan_unit_test ${LIBRARIES} )
```

```
set (LIBRALIES 
       ${OpenCV_LIBS}
       ${gtest} 
       ${gtest_main}
       -lpthread)
```

```

---

cool